src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -146,17 +151,19 @@ public override Expression Visit(Expression node)
                     }
 
                     // Inspect the visited node
-                    this.Inspect();
+                    this.InspectAsync();
```

Comment:
Yeah, we will need to add .result here, but this makes no sense to make method aync.